src/liballoc/heap.rs

```diff
@@ -44,6 +45,84 @@ fn check_size_and_alignment(size: usize, align: usize) {
                   align);
 }
 
+#[derive(Copy, Clone, Default, Debug)]
+pub struct HeapAllocator;
+
+unsafe impl Allocator for HeapAllocator {
+    unsafe fn alloc(&mut self, layout: Layout) -> Result<*mut u8, AllocErr> {
```

Comment:
(since I am now planning to remove the type aliases in response to @alexcrichton feedback, the question of which to use here becomes irrelevant. That is, unless you want to argue in favor of the type aliases, yay!)